### PR TITLE
Update view_channel.php

### PR DESCRIPTION
**Pull request by aliggie**
Useful change to create unique meta descriptions for the view_channel page (see global_header)

### CODE REVIEW

**Comment by MacWarrior** [comment]
Hi !

Testing modifications, please don't add new commits :)

Thanks by advance,
Best regards,
MacWarrior.

File: upload/sitemap.php [hunk 1]
```diff
@@ -2,50 +2,56 @@
...
+	<url>
+      		<loc><?=BASEURL ?></loc>
+	</url>
 <!-- by fjulio ( tangi @ clipbucket dev. ) -->
 <?php
     foreach($videos as $video) {
 ?>
 <url>
-<loc><?php echo video_link($video); ?></loc>
+<loc><?=video_link($video)?></loc>
```
Comment (MacWarrior) [inline upload/sitemap.php:21]: Be careful with codestyle :
- Don't use short php tag
- Don't forget ending semicolon

File: upload/styles/cb_28/layout/blocks/channels.html [hunk 2]
```diff
@@ -54,7 +54,7 @@ <h2>{$featured.username}</h2>
 {assign var='user_cover_tune' value=$userquery->getCover($user.userid)}
 
 <div class="channel-holder">
-	<a href="{$userquery->profile_link($user)}" id="user-{$user.userid}" class="user-block">
+	<a href="{$userquery->profile_link($user)}" id="user-{$user.userid}" class="user-block" title="{$user.username}">
```
Comment (MacWarrior) [inline upload/styles/cb_28/layout/blocks/channels.html:57]: New (And also old but not yet secured) content editable by users in HTML have to be secured against XSS injection with `display_clean` function

File: upload/styles/cb_28/layout/channels.html [hunk 3]
```diff
@@ -52,13 +52,16 @@
...
-									<a href="{link name=time sort=$sort type=channels}&timing=time" {if $smarty.get.time==$sort} class="selected"{/if}>{if $smarty.get.time==$sort}<i class="icon-tick"></i>{/if}{$name}</a>
+									<a href="{link name=time sort=$sort type=channels}&timing=time" {if $smarty.get.time==$sort} class="selected"{/if}>{if $smarty.get.time==$sort}<i class="icon-tick"></i>{/if}{$name}{if $smarty.get.time==$sort}{$selectedtime=$name}{/if}</a>  
 								</li>
 							{/foreach}
 						  </ul>
 					</div>
 				</div>
 			</div>
+			
+			<h1>{$catname} | {$selectedview} | {$selectedtime} | Channels</h1>
```
Comment (MacWarrior) [inline upload/styles/cb_28/layout/channels.html:63]: Don't forget to use translations when they already exists 😄

File: upload/styles/cb_28/theme/css/all.css [hunk 4]
```diff
@@ -5,7 +5,8 @@
   font-weight: normal;
   font-style: normal; }
 @font-face {
-  font-family: 'source_sans_prolight';
+  font-family: 'source_sans_prolightf-popular
+	  ';
```
Comment (MacWarrior) [inline upload/styles/cb_28/theme/css/all.css:9]: Little typo

File: upload/styles/cb_28/layout/global_header.html [hunk 5]
```diff
@@ -49,8 +55,8 @@
 		<link rel="stylesheet" href="{$theme}/css/all.css?cache={$cache_buster}">
 		{*/if*}
 
-		<script src="{$theme}/js/jquery-1.11.3.min.js"></script>
-		<script src="/js/ui/jquery-ui-1.12.1.min.js"></script>
+		<script src="{$theme}/js/jquery-3.5.1.min.js"></script>
```
Comment (MacWarrior) [inline upload/styles/cb_28/layout/global_header.html:58]: You can't simply replace jquery 1.X to 3.X, without expecting code incompatibility 😉
Front office is broken with this modification : many JS error to fix

File: upload/videos.php [hunk 6]
```diff
@@ -58,9 +58,27 @@
...
 	}
+
+	if (!$_GET['seo_cat_name']) $catname='All Categories';
+	else $catname = str_replace("-", " ", $_GET['seo_cat_name']);
+	
+	$subtitle = $catname .' &raquo; '. $sort_name .' &raquo; '. $sort_time .' &raquo; Videos';
+
+	assign('catname', $catname);
+
 	subtitle(lang($subtitle));
```
Comment (MacWarrior) [inline upload/videos.php:82]: `lang` function can't be used on a dynamic string ; it won't work as intended

File: upload/view_channel.php [hunk 7]
```diff
@@ -81,8 +81,10 @@
                     }
                 }
             }
-
-            subtitle(sprintf(lang('user_s_channel'),$udetails['username']));
+		
+	    $user_s_channel = sprintf(lang(''),$udetails['username']);
```
Comment (MacWarrior) [inline upload/view_channel.php:85]: Be carefull, `lang` var is missing

File: upload/view_channel.php [hunk 8]
```diff
@@ -81,8 +81,10 @@
                     }
                 }
             }
-
-            subtitle(sprintf(lang('user_s_channel'),$udetails['username']));
+		
+	    $user_s_channel = sprintf(lang(''),$udetails['username']);
+            subtitle($user_s_channel);
```
Comment (MacWarrior) [inline upload/view_channel.php:86]: Unessary tab 😄

**Comment by MacWarrior** [comment]
Ok, it's pretty new for me to work on pull requests on github
Even if the pull request is marked as "merged", you can continue to discuss here, I'll follow new messages

Next one will be better handled 😄

Thanks for your work,
Best Regards,
MacWarrior.

File: upload/search_result.php [hunk 9]
```diff
@@ -96,10 +96,13 @@
...
 	{
 		$squery = get('query');
+		$squery = strtolower($squery);
+		$squery = ucwords($squery);
+		assign('search_result', $squery);
 		if ($squery == 'clipbucket') {
 			subtitle("Awesomeness...!!");
 		} else {
-			subtitle($search->search_type[$type]['title'].' : '.display_clean(get('query')));
+			subtitle($search->search_type[$type]['title'].' : '.$squery);
```
[the remark this answers is not in the document]
Comment (aliggie) [reply upload/search_result.php:105]: Thanks

File: upload/channels.php [hunk 10]
```diff
@@ -57,10 +57,27 @@
...
 	$pages->paginate($total_pages,$page,$link,$extra_params,$tag);
-	if (!$subtitle) {
-		$subtitle = 'channels';
+	
+	$sorting_links=sorting_links();
+	$time_links=time_links();
+	
+	foreach($sorting_links as $na_me => $name){
+			if ($na_me == $_GET['sort']) 
+				$sort_name = $sorting_links[$na_me];
```
[the remark this answers is not in the document]
Comment (aliggie) [reply upload/channels.php:66]: Thanks

File: upload/sitemap.php (hunk 1 above)
Comment (aliggie) [reply upload/sitemap.php:21]: Thanks
Why don't you like short php tag? :D

File: upload/styles/cb_28/layout/blocks/channels.html (hunk 2 above)
Comment (aliggie) [reply upload/styles/cb_28/layout/blocks/channels.html:57]: Thanks. I just saw the function inserted in functions.php, essential function against xss. I will keep it in mind!

File: upload/styles/cb_28/layout/channels.html (hunk 3 above)
Comment (aliggie) [reply upload/styles/cb_28/layout/channels.html:63]: Thanks

Comment (aliggie) [reply upload/styles/cb_28/layout/channels.html:63]: Thanks

Comment (aliggie) [reply upload/styles/cb_28/layout/channels.html:63]: For example like this?
{lang($selectedview)}
Thanks

File: upload/styles/cb_28/layout/global_header.html (hunk 5 above)
Comment (aliggie) [reply upload/styles/cb_28/layout/global_header.html:58]: You are right. Thanks

File: upload/videos.php (hunk 6 above)
Comment (aliggie) [reply upload/videos.php:82]: Thanks

File: upload/view_channel.php (hunk 7 above)
Comment (aliggie) [reply upload/view_channel.php:85]: Thanks

**Comment by aliggie** [comment]
I'm currently testing on ClipBucket 4.1. As soon as possible I will update to the latest version 5.3.1.

Do I have to follow any special measures to update from the ClipBucket version 4.1 to 5.3.1?
So we will work on the same version

File: upload/sitemap.php (hunk 1 above)
Comment (MacWarrior) [reply upload/sitemap.php:21]: Hi,

Some PHP configs don't allow short tags, and if I'm correct, short tags are deprecated in PHP 7.4 and will be removed in PHP 8

File: upload/styles/cb_28/layout/channels.html (hunk 3 above)
Comment (MacWarrior) [reply upload/styles/cb_28/layout/channels.html:63]: AS JS is broken on current test branch, I haven't been hable to test all modifications you've done.

But if I'm right, $selectedview contains the value and not the key ; and the value is already translated in the original function : /includes/functions.php : function sorting_links()

So no usage of the `lang` function here

**Comment by MacWarrior** [comment]
Hi,

All you have to do : 
- Get the latest sources : as your willing to work, you might want to work on DEV version, so use git to checkout the repo ; remove the 4.1 directory and replace with 5.3.1 directory
- Remove the file /files/temp/install.me
- Put your old /includes/dbconnect.php on the new directory
- Manually apply /cb_install/sql/upgrade_from_original.sql file to your DB (after remplacing {tbl_prefix} with your own prefix)

There is an existing upgrade system, but it is outdated ant it won't work on latest version ; I plan do refactor it but it's not yet on the top of the todo list 😄

Don't forget to update your fork with the test branch

File: upload/styles/cb_28/layout/channels.html (hunk 3 above)
Comment (aliggie) [reply upload/styles/cb_28/layout/channels.html:63]: So you were referring to the word "Channels"
{lang('channels')}

Comment (MacWarrior) [reply upload/styles/cb_28/layout/channels.html:63]: Hi

That's it ; check my last commit on test branch, I've already done this translation 😄

**Comment by MacWarrior** [comment]
Hi,

Do you have any news about JS compatibility after jQuery upgrade ?
I will definitely work on it if you don't want / can't continue in this path, it's just... not my priority 😄 

Best regards,
MacWarrior.

**Comment by MacWarrior** [comment]
PS : To continue here, you should make a new form of the test branch which has been updated after your merge request

**Comment by aliggie** [comment]
> Hi,
> 
> All you have to do :
> 
> * Get the latest sources : as your willing to work, you might want to work on DEV version, so use git to checkout the repo ; remove the 4.1 directory and replace with 5.3.1 directory
> * Remove the file /files/temp/install.me
> * Put your old /includes/dbconnect.php on the new directory
> * Manually apply /cb_install/sql/upgrade_from_original.sql file to your DB (after remplacing {tbl_prefix} with your own prefix)
> 
> There is an existing upgrade system, but it is outdated ant it won't work on latest version ; I plan do refactor it but it's not yet on the top of the todo list 😄
> 
> Don't forget to update your fork with the test branch

Thanks, in the next few days I will try to update to the latest version. So I can actively collaborate

**Comment by aliggie** [comment]
> Hi,
> 
> Do you have any news about JS compatibility after jQuery upgrade ?
> I will definitely work on it if you don't want / can't continue in this path, it's just... not my priority 😄
> 
> Best regards,
> MacWarrior.

It doesn't work and I reset it as originally. I will see what was wrong

**Pull request merged**